src/test/AludelV3.t.sol

```diff
@@ -296,4 +270,359 @@ contract AludelV3Test is Test {
+
+        IAludelV3.RewardSchedule[] memory schedules = aludel.getAludelData().rewardSchedules;
+        assertEq(schedules.length, 1);
+        assertEq(schedules[0].duration, duration);
+        assertEq(schedules[0].shares, reward * BASE_SHARES_PER_WEI);
+        assertEq(schedules[0].start, block.timestamp);
+    }
+
+    // function test_when_fund_params_are_valid_then_adds_new_schedule(uint256 reward, uint256 duration) public callerOwner {
+    function test_when_fund_succeeds_then_adds_new_schedule(uint256 reward, uint256 duration) public callerOwner {
+        reward = bound(reward, 1, MAX_REWARD);
+        duration = bound(duration, 1, 1000 days);
+        uint256 iterations = type(uint8).max;
+
+        console2.log(reward, duration, iterations);
+        Utils.fundMockToken(admin.addr(), rewardToken, reward * iterations);
+        rewardToken.approve(address(aludel), reward * iterations);
+
+        assertEq(aludel.getAludelData().rewardSchedules.length, 0);
+
+        aludel.fund(reward, duration);
+
+        IAludelV3.RewardSchedule[] memory schedules = aludel.getAludelData().rewardSchedules;
+        assertEq(schedules.length, 1);
+        assertEq(schedules[0].duration, duration);
+        assertEq(schedules[0].shares, reward * BASE_SHARES_PER_WEI);
+        assertEq(schedules[0].start, block.timestamp);
+    }
+
+    // test_when_fund_params_are_valid_and_is_already_funded_then_adds_new_schedule
+    // test_when_fund_params_are_valid_and_is_already_funded_but__then_adds_new_schedule
+
+    function test_when_fund_succeeds_then_transfer_funds_to_reward_pool(uint256 reward, uint256 duration)
+        public
+        callerOwner
+    {
+        reward = bound(reward, 1, MAX_REWARD);
+        duration = bound(duration, 1, 1000 days);
+
+        Utils.fundMockToken(admin.addr(), rewardToken, reward);
+
+        rewardToken.approve(address(aludel), reward);
+
+        aludel.fund(reward, duration);
+
+        // the reward pool should have the reward tokens
+        assertEq(rewardToken.balanceOf(aludel.getAludelData().rewardPool), reward);
+        // the admin should not have any reward tokens left
+        assertEq(rewardToken.balanceOf(admin.addr()), 0);
+    }
+
+    function test_when_fund_succeeds_then_emits_event(uint256 reward, uint256 duration) public callerOwner {
+        reward = bound(reward, 1, MAX_REWARD);
+        duration = bound(duration, 1, 1000 days);
+
+        Utils.fundMockToken(admin.addr(), rewardToken, reward);
+
+        rewardToken.approve(address(aludel), reward);
+
+        vm.expectEmit(true, true, true, true, address(aludel));
+        emit AludelFunded(reward, duration);
+
+        aludel.fund(reward, duration);
+    }
+
+    function test_when_fund_succeeds_then_single_fund_mints_shares(uint256 reward, uint256 duration)
+        public
+        callerOwner
+    {
+        reward = bound(reward, 1, MAX_REWARD);
+        duration = bound(duration, 1, 1000 days);
+
```

Comment:
why all the whitespace?

Reply:
I personally don't mind it.
if neither forge fmt nor lint-sol cries about it, I'll just go with it.

Reply:
for me the whitespace in source code contributes to how I understand it, I expect empty lines to have a similar function to the paragraph separation in natural languages, and when I see two emtpy lines together I expect some big context switch, such as in sections in a book. (or, say, two different contracts, or public vs internal functions of the same contract)

Having the whitespace mean nothing at all (such as having 1-2 emtpy lines between code that accomplishes the same thing) adds a bit of cognitive noise to processing code for me.

this, for example, is how this function makes most sense to me:
```
        reward = bound(reward, 1, MAX_REWARD);
        duration = bound(duration, 1, 1000 days);
        Utils.fundMockToken(admin.addr(), rewardToken, reward);
        rewardToken.approve(address(aludel), reward);

        aludel.fund(reward, duration);
        assertEq(aludel.getAludelData().rewardSharesOutstanding, reward * BASE_SHARES_PER_WEI);
```